src/core/mac/mac.cpp

```diff
@@ -886,8 +912,24 @@ void Mac::HandleBeginTransmit(Timer &aTimer)
 
 void Mac::HandleBeginTransmit(void)
 {
-    Frame &sendFrame(*mTxFrame);
+    Frame *theFrame = NULL;
     otError error = OT_ERROR_NONE;
+    bool applyTransmitSecurity = true;
+
+    switch (mOperation)
+    {
+    case kOperationTransmitOutOfBandFrame:
+        theFrame = mOobFrame;
+        break;
+
+    default:
+        theFrame = mTxFrame;
+        break;
+    }
+
+    assert(theFrame != NULL);
+
+    Frame &sendFrame(*theFrame);
```

Comment:
nit: can we avoid variable declaration in middle of method? 

Possible solution to  define a private method to get the frame:
```
Frame &MacGetTxFrame(void)  
{
    Frame *frame = NULL;

    switch (mOperation)
     {
     case kOperationTransmitOutOfBandFrame:
         frame = mOobFrame;
         break;
 
     default:
         frame = mTxFrame;
         break;
     }

    return *frame;
}
````
And then from `Mac::HandleBeginTransmit()` we do 
```
    Frame &sendFrame = GetTxFrame();